src/snmalloc/backend_helpers/smallbuddyrange.h

```diff
@@ -204,7 +213,8 @@ namespace snmalloc
 
       CapPtr<void, ChunkBounds> alloc_range(size_t size)
       {
-        SNMALLOC_ASSERT(size < MIN_CHUNK_SIZE);
+        if (size >= MIN_CHUNK_SIZE)
+          return parent.alloc_range(size);
```

Comment:
I recall us going back and forth on this previously, with the idea that the caller should skip the `SmallBuddy` ranges if they needed something larger.  This fallback requires that the return type of the `SmallBuddy` match that of its parent, which might not, in general, be true: while a `LargeBuddy` "must" operate on `CapPtr`s intended to be large enough to include whole `PageMap` `MetaEntry`s worth of space, as it uses those for its metadata, the `SmallBuddy`, which uses in-band metadata, could have more tightly bounded types.

I suppose we're not actually taking advantage of that right now, tho'... but is there reason not to ask for the Range's `parent` in the caller instead?

Reply:
So ultimately we need a branch somewhere in `alloc_meta_data`.  This would need to determine how large the request is and then which range to send it too.  The code in `SmallBuddyRange::alloc_with_leftover`, returns the slop at the end of the allocation.  This could be useful for large meta data allocations as well as small, so might want to be duplicated, though not essential.

Reply:
Following Teams call, I have added a test to ensure future changes don't break the Verona use case.

We will leave this design as is, but will revisit after @nwf-msr investigates a fine granularity of permissions.